Drop syncable from incremental CV export

maximiliankolb opened the pull request:
If you want to perform an incremantal content view export in a syncable format, see "proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc".

* Sort arguments in alphabetical order
* Reduce diff between two very similar procedures
* Use "find" over "ls" but drop pkg extension because exports also work for file-type content, not just Yum content

## two notes

* I would be happy with the wording from either file
* I would also be happy to drop one procedure and instead add a hint about the `--syncable` flag.

* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.10/Katello 4.12
* [x] Foreman 3.9/Katello 4.11 (planned Satellite 6.15)

adamlazik1 commented:
I'm in on removing the syncable procedure and just hinting at the syncable flag in the other one, what do you think @asteflova @apinnick?

apinnick commented:
> I'm in on removing the syncable procedure and just hinting at the syncable flag in the other one, what do you think @asteflova @apinnick?

@adamlazik1  You mean you would not include the procedure section in [Exporting... syncable format](https://theforeman-foreman-documentation-preview-pr-2863.surge.sh/nightly/Managing_Content/index-satellite.html#Exporting_a_Repository_in_a_Syncable_Format_content-management)? Makes sense. I agree. 

This Hammer command is repeated in other modules. It might make sense to have only one procedure with all the flags documented in a table or with callouts, depending on how many there are. All these code block examples seem like overkill.

apinnick commented on guides/common/modules/proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc:
```diff
@@ -7,7 +7,7 @@ The size of the exported content view versions depends on the number of products
 Content view versions that have multiple {RHEL} trees can occupy several gigabytes of space on {ProjectServer}.
 
-In such cases, you can use *Incremental Export* to export only pieces of content that changed since the previous export.
+In such cases, you can create an incremental export which contains only pieces of content that changed since the previous export.
```
```suggestion
In such cases, you can create an incremental export which contains only pieces of content that have changed since the last export.
```

maximiliankolb replied:
Applied (twice)

maximiliankolb commented:
@adamlazik1 @apinnick Is it OK to do this in a follow-up PR?

adamlazik1 commented:
Sure, fine with me.

adamlazik1 commented on guides/common/modules/proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc:
```diff
@@ -19,13 +19,13 @@ The example below targets the version `2.0` for export because the version `1.0`
 [options="nowrap", subs="+quotes,verbatim,attributes"]
 ----
-# find /var/lib/pulp/exports/Default_Organization/view/2.0/2023-03-02T14-19-50-05-00/ -name "*.{client-pkg-ext}"
+# find /var/lib/pulp/exports/_My_Organization_/view/2.0/2024-03-02T14-19-50-05-00/
```
Do we want to include specific date here? Or maybe do we want to go with something more broad like `/var/lib/pulp/exports/_My_Organization_/_My_Content_View_/_My_Content_View_Version_/`?

Additionally, since we've recently been using the `Next steps` subsection, I feel like this could be a good candidate.

maximiliankolb commented:
I have dropped another almost identical procedure and introduced a "syncable exports"-snippet. I think we can squash these commits when merging.

Future TODO: It would be nice to look at all procedures in `guides/common/assembly_synchronizing-content-between-servers.adoc`; maybe we can simplify things even more.

maximiliankolb commented on guides/common/modules/snip_syncable-exports.adoc:
```diff
@@ -0,0 +1,7 @@
++
+If you want to create a syncable export, add `--format=syncable`.
+By default, {Project} creates an importable export.
++
```
@chris1984 I am a bit unsure how to interpret the output from hammer --help:

````
$ hammer content-export incremental version --help
 --format ENUM                                  Export formats.Choose syncable if the exported content needs to be in a yum
                                                format. This option is only available for yum, file repositories. Choose
                                                importable if the importing server uses the same version  and exported content
                                                needs to be one of yum, file, ansible_collection, docker, deb repositories.
                                                Possible value(s): 'syncable', 'importable'
````

Can you please shed some light; maybe ACK my summary?

* syncable > provide from an arbitrary webserver over HTTP to another Foreman server? (basically a "custom CDN" as mention in [10.4. Exporting the library environment in a syncable format](https://docs.theforeman.org/nightly/Managing_Content/index-katello.html#Exporting_the_Library_Environment_in_a_Syncable_Format_content-management)
* importable > create an archive, transfer to disk of another Foreman server; then import?

adamlazik1 commented on guides/common/modules/proc_exporting-a-content-view-version-incrementally.adoc:
```diff
@@ -18,14 +18,18 @@ The example below targets the version `2.0` for export because the version `1.0`
+[options="nowrap", subs="+quotes,verbatim,attributes"]
 ----
-# ls -lh /var/lib/pulp/exports/_My_Organization_/view/2.0/2021-02-25T21-45-34-00-00/
+# find /var/lib/pulp/exports/_My_Organization_/view/_My_Content_View_Version_/
```
Unable to verify this, but I am guessing this is how the exports are stored? Consider this non-blocking if you are unable to verify as well.
```suggestion
# find /var/lib/pulp/exports/_My_Organization_/_My_Content_View_/_My_Content_View_Version_/
```

maximiliankolb replied:
Tested on Foreman 3.9. The name depends on the "--format" option:

````
# hammer content-export complete repository --name "CentOS Client 8 stream" --organization-id 1 --product "CentOS Client"
[.....................................................................................................................................................................................................] [100%]
Generated /var/lib/pulp/exports/Example/Export-CentOS_Client_8_stream-5/1.0/2024-03-08T09-17-27-01-00/metadata.json

# hammer content-export complete repository --name "CentOS Client 8 stream" --organization-id 1 --product "CentOS Client" --format=syncable
[.....................................................................................................................................................................................................] [100%]
Generated /var/lib/pulp/exports/Example/Export-SYNCABLE-CentOS_Client_8_stream-5/1.0/2024-03-08T09-18-21-01-00/metadata.json
````

Org is called "Example".

adamlazik1 commented on guides/common/modules/proc_exporting-a-content-view-version-incrementally.adoc:
```diff
@@ -1,13 +1,13 @@
-In such cases, you can create an incremental export which contains only pieces of content that changed since the previous export.
+In such cases, you can create an incremental export which contains only pieces of content that have changed since the last export.
 Incremental exports typically result in smaller archive files than the full exports.
 
 The example below targets the version `2.0` for export because the version `1.0` was exported previously.
```
```suggestion
```

adamlazik1 commented on guides/common/modules/proc_exporting-a-content-view-version-incrementally.adoc:
```diff
@@ -18,14 +18,18 @@ The example below targets the version `2.0` for export because the version `1.0`
 ----
 # hammer content-export incremental version \
---content-view="_Content_View_Name_" \
+--content-view="_My_Content_View_Name_" \
```
```suggestion
--content-view="_My_Content_View_" \
```

maximiliankolb replied:
Applied.

adamlazik1 commented on guides/common/modules/proc_exporting-a-content-view-version-incrementally.adoc:
```diff
@@ -18,14 +18,18 @@ The example below targets the version `2.0` for export because the version `1.0`
 --organization="_My_Organization_" \
---version=2.0 \
---format=syncable
+--version=2.0
```
```suggestion
--version="_My_Content_View_Version_"
```

adamlazik1 reviewed (commented):
I like where this PR is going. Some more suggestions.

maximiliankolb commented on guides/common/modules/proc_exporting-a-content-view-version-incrementally.adoc:
```diff
@@ -18,14 +18,18 @@ The example below targets the version `2.0` for export because the version `1.0`
+--version=2.0
 ----
+include::snip_syncable-exports.adoc[]
 . Optional: View the exported content view:
```
Moved.

maximiliankolb commented on guides/common/modules/proc_exporting-the-library-environment-incrementally.adoc:
```diff
@@ -1,33 +1,29 @@
-[id="Exporting_Library_Incrementally_{context}"]
-= Exporting the library environment incrementally
+[id="Exporting-the-Library-environment-incrementally_{context}"]
```
Applied the first suggestion.

maximiliankolb commented on guides/common/modules/proc_exporting-the-library-environment-incrementally.adoc:
```diff
@@ -1,33 +1,29 @@
+= Exporting the Library environment incrementally
 
-Exporting Library content can be a very expensive operation in terms of system resources.
+Exporting Library content can be a very costly operation in terms of system resources.
```
OK, reverted. FYI: this was not my own idea but part of "merging" syncable and non-syncable procedures.

maximiliankolb commented on guides/common/modules/proc_exporting-the-library-environment-incrementally.adoc:
```diff
@@ -1,33 +1,29 @@
+--organization="_My_Organization_"
 ----
+include::snip_syncable-exports.adoc[]
 . Optional: View the exported data:
```
Applied.

maximiliankolb reviewed (commented):
Thanks Adam; applied all feedback.

Should I use an ordered or unordered list under "next steps"?

asteflova commented:
> Should I use an ordered or unordered list under "next steps"?

Unordered (FYI, that's what the [procedure module template](https://github.com/redhat-documentation/modular-docs/blob/1848b01ae780fcb2cef45a694a0634db76c84937/modular-docs-manual/files/TEMPLATE_PROCEDURE_doing-one-procedure.adoc?plain=1#L60) says).

adamlazik1 reviewed (commented):
Just one little thing, rest LGTM.

maximiliankolb commented on guides/common/modules/proc_exporting-a-content-view-version-incrementally.adoc:
```diff
@@ -1,31 +1,33 @@
 = Exporting a content view version incrementally
 
-Exporting complete versions can be a very expensive operation in terms of system resources.
+Exporting complete content view versions can be a very costly operation in terms of system resources.
```
Fixed twice.

maximiliankolb commented:
Thanks @adamlazik1 ; ready for final ACK :)

adamlazik1 reviewed (approved):
LGTM

maximiliankolb commented:
Merged to "master" and cherry-picked:
   4ee89c9011..b1dfd73f51  3.10 -> 3.10
   67cf926c93..9f47edef2e  3.9 -> 3.9
